src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Unix.cs

```diff
@@ -64,6 +64,12 @@ public static int GetProcessIdFromHandle(SafeProcessHandle processHandle)
         /// <returns>true if the machine is remote; false if it's local.</returns>
         public static bool IsRemoteMachine(string machineName)
         {
+            if (machineName == null)
+                throw new ArgumentNullException(nameof(machineName));
+
+            if (machineName.Length == 0)
+                throw new ArgumentException(SR.Format(SR.InvalidParameter, nameof(machineName), machineName));
+
```

Comment:
Thanks. We try to keep this kind of validation consolidated rather than duplicated. Could you factor it out into the shared file, and then have that code call into these helpers renamed as IsRemoteProcessCore?